### PR TITLE
MRKT-32-Single-Item-Buy-Function

### PR DESCRIPTION
**Pull request by dmkirshon**
Add (To Item Page):

- Buy Function UI with initial Formik layout
- Tooltip for buying Stream Tokens
- Buy Button
- Skeleton updates

image

### CODE REVIEW

File: apps/marketplace/src/app/item/[songId]/page.tsx [hunk 1]
```diff
@@ -86,6 +92,93 @@ const SingleSong: FunctionComponent<SingleSongProps> = ({ params }) => {
...
+                            title={
+                              "The number of Stream Tokens correlates directly " +
+                              "with the percentage of Streaming royalties you " +
+                              "can acquire and the total price of the bundle. " +
+                              "For example 1 token is worth = 0.0000001% of " +
+                              "total royalties, and costs ‘Ɲ3.0‘."
+                            }
+                          >
+                            <IconButton sx={ { padding: 0 } }>
+                              <InfoIcon sx={ { color: theme.colors.grey100 } } />
```
Comment (dmkirshon) [inline apps/marketplace/src/app/item/[songId]/page.tsx]: NEWM Studio uses a question mark icon consistently, per Marketplace Figma a "i" info icon is used. I left a comment in Figma for design to review, if alignment is needed we will update in future PR.

File: apps/marketplace/src/app/item/[songId]/page.tsx [hunk 2]
```diff
@@ -86,6 +92,93 @@ const SingleSong: FunctionComponent<SingleSongProps> = ({ params }) => {
                 { songData?.artist.firstName } { songData?.artist.lastName }
               </Typography>
             </Stack>
+
+            <Formik
+              initialValues={ { streamTokens: 1 } }
+              onSubmit={ () => {
+                return;
```
Comment (escobarjonatan) [inline apps/marketplace/src/app/item/[songId]/page.tsx:99]: so the way i see it is... If user wants to buy..... DON'T!

**Review by escobarjonatan — APPROVED** [review]

**Review by scandycuz — APPROVED** [review]
LGTM

**Comment by scandycuz** [comment]
One thing that would be good to do is move the top "purchase song-tokens" section of this page to it's own component, with it's own loading skeleton as part of that component, so that the rest of the page's loading skeletons will appear during loading as well. That can be done in a future PR though, potentially as part of the work to add the "meta-data" section to the page.

**Pull request merged**